lib/webapp.js

```diff
@@ -134,6 +151,41 @@ DataserviceContext.prototype = {
   makeErrorObject: zluxUtil.makeErrorObject
 };
 
+function createDataserviceStorage(pluginId) {
+  
+  return process.clusterManager.getStorageAll(pluginId).then(function (clusterStorage) {
+
+    /* Get the whole plugin storage */
+    clusterStorage.getStorageAll = function () {
```

Comment:
Dont modify clusterStorage here, just create a brand new object with these functions and return it... basically the object returned by getStorageAll here is pointless to you here.